ui/packages/catalog-ui-search/src/main/webapp/js/EventSourceUtil.js

```diff
@@ -0,0 +1,29 @@
+var EventSource = require('eventsource')
+
+const REQUEST_HEADER = 'XMLHttpRequest'
+const HEADERS = {
+  'X-Requested-With': REQUEST_HEADER,
+}
+
+var sources = []
+
+module.exports = {
+  createEventListener(type, handlers) {
+    const { onMessage } = handlers
+    if (sources.length != 0) {
+      sources[0].addEventListener(type, event => {
```

Comment:
will sources only ever have one element?

Reply:
yes - refactoring this